### ISSUE TITLE
Generating Pods project error: undefined method `native_target' of Amplitude-iOS 

### ISSUE THREAD

**Issue by ZackLeonardo** [expo/expo#2401]
### Environment

Environment:
  OS: macOS High Sierra 10.13.6
  Node: 10.8.0
  Yarn: 0.27.5
  npm: 6.2.0
  Watchman: 4.9.0
  Xcode: Xcode 10.0 Build version 10A255
  Android Studio: 3.0 AI-171.4443003

Packages: (wanted => installed)
  expo: ^30.0.0 => 30.0.1
  react: 16.3.1 => 16.3.1
  react-native: https://github.com/expo/react-native/archive/sdk-30.0.0.tar.gz => 0.55.4

Diagnostics report:
  https://exp-xde-diagnostics.s3.amazonaws.com/zackleonardo-bc4d1eee-f676-4f5c-b037-1e6c45986b9e.tar.gz

### Steps to Reproduce

1. $exp detach
2.$cd ios 
3.$pod install

Installing boost-for-react-native (1.63.0)
Installing glog (0.3.4)
Installing lottie-ios (2.5.0)
Installing yoga (0.55.4.React)
Generating Pods project
[!] An error occurred while processing the post-install hook of the Podfile.

undefined method `native_target' for <Pod::PodTarget name=Amplitude-iOS >:Pod::PodTarget

/Users/zoudeyi/RNProjects/react-native-chat/ios/Podfile:123:in `block (4 levels) in from_ruby'
/Users/zoudeyi/RNProjects/react-native-chat/ios/Podfile:107:in `each'
/Users/zoudeyi/RNProjects/react-native-chat/ios/Podfile:107:in `block (3 levels) in from_ruby'
/Library/Ruby/Gems/2.3.0/gems/cocoapods-core-1.6.0.beta.1/lib/cocoapods-core/podfile.rb:179:in `post_install!'
/Library/Ruby/Gems/2.3.0/gems/cocoapods-1.6.0.beta.1/lib/cocoapods/installer.rb:685:in `run_podfile_post_install_hook'
/Library/Ruby/Gems/2.3.0/gems/cocoapods-1.6.0.beta.1/lib/cocoapods/installer.rb:673:in `block in run_podfile_post_install_hooks'
/Library/Ruby/Gems/2.3.0/gems/cocoapods-1.6.0.beta.1/lib/cocoapods/user_interface.rb:145:in `message'
/Library/Ruby/Gems/2.3.0/gems/cocoapods-1.6.0.beta.1/lib/cocoapods/installer.rb:672:in `run_podfile_post_install_hooks'
/Library/Ruby/Gems/2.3.0/gems/cocoapods-1.6.0.beta.1/lib/cocoapods/installer.rb:197:in `block in generate_pods_project'
/Library/Ruby/Gems/2.3.0/gems/cocoapods-1.6.0.beta.1/lib/cocoapods/user_interface.rb:64:in `section'
/Library/Ruby/Gems/2.3.0/gems/cocoapods-1.6.0.beta.1/lib/cocoapods/installer.rb:194:in `generate_pods_project'
/Library/Ruby/Gems/2.3.0/gems/cocoapods-1.6.0.beta.1/lib/cocoapods/installer.rb:125:in `install!'
/Library/Ruby/Gems/2.3.0/gems/cocoapods-1.6.0.beta.1/lib/cocoapods/command/install.rb:48:in `run'
/Library/Ruby/Gems/2.3.0/gems/claide-1.0.2/lib/claide/command.rb:334:in `run'
/Library/Ruby/Gems/2.3.0/gems/cocoapods-1.6.0.beta.1/lib/cocoapods/command.rb:52:in `run'
/Library/Ruby/Gems/2.3.0/gems/cocoapods-1.6.0.beta.1/bin/pod:55:in `<top (required)>'
/usr/local/bin/pod:22:in `load'
/usr/local/bin/pod:22:in `<main>'

**Comment by ZackLeonardo** [comment]
pod 1.5.3 is OK.

**Comment by kokorolee** [comment]
You can downgrade cocoapods to v1.5.3:
* `sudo gem uninstall cocoapods`
* say `Y` when it asks whether to remove executables
* `sudo gem install cocoapods -v 1.5.3`

**Comment by macojaune** [comment]
I have the same error with cocoapods v1.5.3 and expo v32

**Comment by akagaeng** [comment]
I resolve this issue as follows.

Actually I summarized above.
Thanks to @ZackLeonardo @kokorolee @macojaune !

### Check pod version

I tried on `v1.6.1` and `v1.5.2`. But it fails. (Mac app version)
So I uninstalled every version of cocoapods. 

### Uninstall preinstalled cocoapod
```
$ gem uninstall cocoapod
```

### install cocoapod v1.5.3

```
$ gem install cocoapods -v 1.5.3
```

### Check pod version again
My cocoapod version is now `v1.5.3`

```
$ pod --version
1.5.3
```

### Install pod
In ios directory,  run pod install.

```
<your project directory>/ios $ pod install
```

### Done!
Finally is it successfully installed!

```
Pod installation complete! 
There are 54 dependencies from the Podfile and 63 total pods installed.
```

**Comment by macojaune** [comment]
Thanks for your answer,
I figured it out already. My problem was that I had severals version of cocoapods installed so the version I was targeting didn't match with the one used.



Le jeu. 4 avr. 2019 à 21:27, akagaeng <notifications@github.com<mailto:notifications@github.com>> a écrit :

I resolve this issue as follows.

Actually I summarized above.
Thanks to @ZackLeonardo<https://github.com/ZackLeonardo> @kokorolee<https://github.com/kokorolee> @macojaune<https://github.com/macojaune> !

Check pod version

I tried on v1.6.1 and v1.5.2. But it fails. (Mac app version)
So I uninstalled every version of cocoapods.

Uninstall preinstalled cocoapod

$ gem uninstall cocoapod


install cocoapod v1.5.3

$ gem install cocoapods -v 1.5.3


Check pod version again

My cocoapod version is now v1.5.3

$ pod --version
1.5.3


Install pod

In ios directory, run pod install.

<your project directory>/ios $ pod install


Done!

Finally is it successfully installed!

Pod installation complete!
There are 54 dependencies from the Podfile and 63 total pods installed.
